src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentConverter.cs

```diff
@@ -24,85 +23,71 @@ public class CoraxDocumentConverter : CoraxDocumentConverterBase
+                        if (BlittableJsonTraverserHelper.TryRead(_blittableTraverser, document, spatialOptions.MethodArguments[0], out var latValue) == false)
+                            continue;
 
-                            value = StaticIndexBase.CreateSpatialField(spatialField, latValue, lngValue);
-                            break;
-                        default:
-                            throw new ArgumentOutOfRangeException($"{spatialOptions.MethodType} is not implemented.");
-                    }
+                        if (BlittableJsonTraverserHelper.TryRead(_blittableTraverser, document, spatialOptions.MethodArguments[1], out var lngValue) == false)
+                            continue;
 
-                    InsertRegularField(indexField, value, indexContext, ref entryWriter, sourceDocument, scope, out var _);
+                        value = AbstractStaticIndexBase.CreateSpatialField(spatialField, latValue, lngValue);
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException($"{spatialOptions.MethodType} is not implemented.");
                 }
-                else if (BlittableJsonTraverserHelper.TryRead(_blittableTraverser, document, indexField.OriginalName ?? indexField.Name, out value))
-                {
-                    InsertRegularField(indexField, value, indexContext, ref entryWriter, sourceDocument, scope, out var _);
-                }
-            }
 
-            if (key != null)
+                InsertRegularField(indexField, value, indexContext,builder, sourceDocument, out var _);
+            }
+            else if (BlittableJsonTraverserHelper.TryRead(_blittableTraverser, document, indexField.OriginalName ?? indexField.Name, out value))
             {
-                Debug.Assert(document.LowerId == null || (key == document.LowerId));
-                scope.Write(string.Empty, 0, id.AsSpan(), ref entryWriter);
+                InsertRegularField(indexField, value, indexContext, builder, sourceDocument,  out var _);
             }
             
-            if (_storeValue)
+            hasFields = true;
+        }
+
+        if (key != null)
+        {
+            Debug.Assert(document.LowerId == null || (key == document.LowerId));
+            builder.Write( 0, string.Empty, id.AsSpan());
+        }
+            
+        if (_storeValue)
+        {
+            unsafe
             {
-                unsafe
+                using (Allocator.Allocate(document.Data.Size, out Span<byte> blittableBuffer))
                 {
-                    using (Allocator.Allocate(document.Data.Size, out Span<byte> blittableBuffer))
-                    {
-                        fixed (byte* bPtr = blittableBuffer)
-                            document.Data.CopyTo(bPtr);
+                    fixed (byte* bPtr = blittableBuffer)
+                        document.Data.CopyTo(bPtr);
```

Comment:
can we dispose document after this?

Reply:
That is coming from the caller, not for us to dispose, no?

Reply:
Ok